notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,548 @@
+ 10. These paramaters will be stored in a dictionary that will be used in a function to plot the original parameters used to simulate the choices (from step 3) along with any recovered parameters from step 9.
+ 11. One variable will be perturbed to simulate a potential external intervention. In this scenario, no other variables will be changes since all covariates are independent.
+ 12. Causal effects will be estimated using the models estimated in step 3 and 9.
+ 13. Plots will be produced to show the distribution of "naive" and "true" causal effects
+
+
+For a realistic causal graph with confounding:
+ 1. Ingest data
+ 2. Manipulate data as needed, the output of this step will be a cleaned dataset
+ 3. Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ 4. Specify the realistic causal graph with desired confounding structures. The causal graph will be saved as a causalgraphicalmodels object and can be saved as .png file
+ 5. The cleaned dataset will be used in a function that estimated the distributions for variables of interest that are nodes without any parents in the causal graph, this function will produce a parameter dictionary containing the type of variables, the best fitted distribution, and that distribution parameters.
+ 6. The cleaned dataset from step 2 will be used to estimate relationships between nodes in the specified causal graph from step 4.
+ 7. The obtained distributions from step 5 will be used to simulate data for nodes without parents in the specified causal graph.
+ 8. Data from step 7, alongside with estimated relationships from step 6 will be used to estimate data for the remaining nodes in the causal graph (with the exception of the utility node)
+ 9. Based on observed availability of allternatives, a function will use the cleaned dataset to simulate an availability matrix for alternatives for the simulated dataset
+ 10. The specified model from step 3 will be used to simulate choices for the simulated dataset.
+ 11. The dataset will be used to estimate a choice model and recover the necessary parameters. These parameters can be stored in a dictionary.
+ 12. Steps 7-11 will be repeated for N times (say 100) and the parameters from each iteration will be stored in a dictionary entry.
+ 13. These paramaters will be stored in a dictionary that will be used in a function to plot the original parameters used to simulate the choices (from step 3) along with any recovered parameters from step 9.
+ 14. One variable will be perturbed to simulate a potential external intervention. In this scenario, descendant children nodes of the perturbed node will also be perturbed following the estimated relationships from step 8.
+ 15. Causal effects will be estimated using the models estimated in step 3 and 121.
+ 16. Plots will be produced to show the distribution of "estimated" and "true" causal effects
```

Comment:
I see what you're doing now. Here are my high level comments: 

- The exercise of causal graph with independent variables only is not necessary and does not contribute to our story. All we're doing there is estimate a model on a simulated dataset and seeing if we recover the parameters of the simulation, which is trivial. That used to be our plan but not anymore. I suggest removing it from this document altogether. 

- In the realistic causal graph, there is no need to repeat steps 7-11 N times, because again this is the exercise of seeing whether you recover your simulation parameters if you use the same model specification. Instead, what we will do is repeat the simulation N times, and apply the perturbation at each simulation, and store the average treatment effect. So I would change the steps to the following:

 7. The obtained distributions from step 5 will be used to simulate data for nodes without parents in the specified causal graph.
 8. Data from step 7, alongside with estimated relationships from step 6 will be used to estimate data for the remaining nodes in the causal graph (with the exception of the utility node)
 9. Based on observed availability of alternatives, a function will use the cleaned dataset to simulate an availability matrix for alternatives for the simulated dataset
 10. The specified model from step 3 will be used to simulate choices for the simulated dataset.
 11. The dataset will be used to estimate a choice model and recover the necessary parameters. (These parameters can be stored in a dictionary, if necessary)
 12. One variable will be perturbed to simulate a potential external intervention. This perturbation will result in two modified datasets: one where only the perturbed variable changes (assuming an independent causal graph), and one where descendant children nodes of the perturbed node will also be perturbed following the estimated relationships from step 8.
 13. Causal effects will be estimated using the model estimated in step 11 and the true model specified in step 3 on both datasets. The model from step 3 will be applied only on the dataset resulting from the perturbation of the "true" causal graph, giving us the true causal effect. The model from step 11 will be applied to both datasets from step 12, resulting in our "naive" and "estimated" causal effects. 
 14. Repeat steps 7-13 N times, keeping the same perturbation, and the estimated causal effects will be stored in a dictionary.
 15. Plots will be produced to show the distribution of "estimated" and "true" causal effects

@timothyb0912

Reply:
I will be adding a brief description of what steps are just procedural vs. what steps help in completing the story of the project.